### PR TITLE
Create Build Evidence.

### PR DESCRIPTION
**Pull request by osaidwtd**

### CODE REVIEW

File: evidence/create_build.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	"strings"
+)
+
+type createEvidenceBuild struct {
+	createEvidenceBase
+	project string
+	build   string
+}
+
+func NewCreateEvidenceBuild(serverDetails *coreConfig.ServerDetails, predicateFilePath string, predicateType string, key string, keyId string,
```
Comment (denys-jfrog) [inline evidence/create_build.go:21]: Optional: as the types are the same, the `string` may be put just once:
```go
func NewCreateEvidenceBuild(serverDetails *coreConfig.ServerDetails, 
	predicateFilePath, predicateType, key, keyId, project, build string) Command {
```

File: evidence/create_build.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	err = c.uploadEvidence(envelope, subject)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *createEvidenceBuild) buildBuildInfoSubjectPath(artifactoryClient artifactory.ArtifactoryServicesManager) (string, error) {
+	build := strings.Split(c.build, ":")
```
Comment (denys-jfrog) [inline evidence/create_build.go:67]: Please, raise the question about the input parameters one more time.
This split means that neither Build Name nor Build Number may contain `:` (this is not the case).
Hence, it means that we do not support Builds with name/number having `:`.
Needs to be confirmed explicitly.

File: evidence/create_build.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	return buildInfoPath + "@" + buildInfoChecksum, nil
+}
+
+func getBuildLatestTimestamp(name string, number string, project string, artifactoryClient artifactory.ArtifactoryServicesManager) (string, error) {
+	buildInfo := services.BuildInfoParams{
+		BuildName:   name,
+		BuildNumber: number,
+		ProjectKey:  project,
+	}
+	res, b, err := artifactoryClient.GetBuildInfo(buildInfo)
```
Comment (denys-jfrog) [inline evidence/create_build.go:94]: Please, rename `b` to something like `ok`.

File: evidence/create_build.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	if !b {
+		errorMessage := fmt.Sprintf("failed to find build, name:%s, number:%s, project: %s", name, number, project)
+		if err != nil {
```
Comment (denys-jfrog) [inline evidence/create_build.go:97]: The order of conditions looks wrong, the `err` should be checked first:
```go
if err != nil {
    return "", err
}
if !ok {
    errorMessage := fmt.Sprintf("failed to find build, name:%s, number:%s, project: %s", name, number, project)
    return "", errorutils.CheckErrorf(errorMessage)
}
```

File: evidence/create_build.go (hunk 3 above)
Comment (osaidwtd) [reply evidence/create_build.go:94]: done

File: evidence/create_build.go (hunk 1 above)
Comment (osaidwtd) [reply evidence/create_build.go:21]: done

File: evidence/create_build.go (hunk 4 above)
Comment (osaidwtd) [reply evidence/create_build.go:97]: done

File: evidence/create_build.go (hunk 2 above)
Comment (osaidwtd) [reply evidence/create_build.go:67]: I talked with Noam, there is a chance to change all other operations to be splitter to --name --version.
will take it in a separate Jira ticket.

**Comment by osaidwtd** [comment]
Create Evidence Build-Info Using CLI

**Pull request merged**